produce_request.go

```diff
@@ -214,6 +212,8 @@ func (r *ProduceRequest) requiredVersion() KafkaVersion {
 		return V0_10_0_0
 	case 3:
 		return V0_11_0_0
+	case 4:
```

Comment:
Shouldn't this be `7`? See [Kafka source](https://github.com/apache/kafka/blob/2c44e77e2f2056ee88b4b9cc8bac1f769748830a/clients/src/main/java/org/apache/kafka/common/requests/ProduceRequest.java#L118-L121).

I think you also need to update [this code](https://github.com/Shopify/sarama/blob/4c93f1f002740d2d8604ba4f82d2c87a4d49a1ba/produce_set.go#L124-L129) to start encoding with that version.

Ultimately you want to test this against a live v2.1.0 Kafka and see if that works.